symengine/dict.cpp

```diff
@@ -182,6 +183,36 @@ int map_uint_mpz_compare(const map_uint_mpz &A, const map_uint_mpz &B)
     return 0;
 }
 
+bool map_uint_Expr_eq(const map_uint_Expr &a,
+        const map_uint_Expr &b)
+{
+    // Can't be equal if # of entries differ:
+    if (a.size() != b.size()) return false;
+    // Loop over keys in "a":
+    for (const auto &p: a) {
+        auto f = b.find(p.first);
+        if (f == b.end()) return false; // no such element in "b"
+        if (p.second != f->second) return false; // values not equal
+    }
+    return true;
+}
+
+
+int map_uint_Expr_compare(const map_uint_Expr &A, const map_uint_Expr &B)
+{
+    if (A.size() != B.size())
+        return (A.size() < B.size()) ? -1 : 1;
+    auto a = A.begin();
+    auto b = B.begin();
+    for (; a != A.end(); ++a, ++b) {
+        if (a->first != b->first)
+            return (a->first < b->first) ? -1 : 1;
```

Comment:
Instead of using the `<` operator here, you can return the value given by the `__cmp__` method